.gitignore

```diff
@@ -1,4 +1,17 @@
 
+# We can honestly just ignore the entire build directory
+build/
+
+# This directory can be ignored too. It contains files for ccls,
+# Which is a C++ language server. It's open source, really fast,
+# and it supports C++20. It also supports semantic highlighting,
+# which really helps me concentrate and reduces distraction for me
+# See: https://github.com/MaskRay/ccls
+.ccls-cache/
```

Comment:
Can `ccls` be used with VSCode? _Should_ it be used for VSCode? I'm a huge fan of semantic highlighting.

Reply:
It has full support for vscode! All the screenshots I've been sending on twitter use semantic highlingting done by ccls in vscode! It also has autocomplete, full linting, and tells you if there are any errors in your code!